goodkey/good_key_test.go

```diff
@@ -299,3 +299,37 @@ func TestRSAStrangeSize(t *testing.T) {
 	test.AssertError(t, err, "expected GoodKey to fail")
 	test.AssertEquals(t, err.Error(), "key size not supported: 4")
 }
+
+func TestCheckPrimeFactorsTooClose(t *testing.T) {
+	// The prime factors of 5959 are 59 and 101. The values a and b calculated
+	// by Fermat's method will be 80 and 21. The ceil of the square root of 5959
+	// is 78. Therefore it takes 3 rounds of Fermat's method to find the factors.
+	n := big.NewInt(5959)
+	test.AssertEquals(t, checkPrimeFactorsTooClose(n, 2), false)
+	test.AssertEquals(t, checkPrimeFactorsTooClose(n, 3), true)
+
+	// The prime factors of this integer are:
+	// p: 12451309173743450529024753538187635497858772172998414407116324997634262083672423797183640278969532658774374576700091736519352600717664126766443002156788367
```

Comment:
Why not do:

```
p, ok := p.SetString("12451309173743450529024753538187635497858772172998414407116324997634262083672423797183640278969532658774374576700091736519352600717664126766443002156788367", 10)
...
q, ok := q.SetString("12451309173743450529024753538187635497858772172998414407116324997634262083672423797183640278969532658774374576700091736519352600717664126766443002156788337", 10)

n.Mul(p, q)
```

Then the test code would make it clear that p*q = n.

Reply:
Good idea, done!